plugins/modules/ssm_parameter.py

```diff
@@ -208,12 +236,37 @@
       description: Parameter version number
       example: 3
       returned: success
+    tags:
```

Comment:
@markuman What if we only return tags as a dictionary rather than a list of dicts and remove tags_dict from the response?

Reply:
@markuman @alinabuzachis To provide a little context, tags_dict was copied over from plugins/modules/secretsmanager_secret.py. I'd be happy to remove the code if you'd like.

Reply:
secretsmanager_secret has some compatibility code in there because it originally returned the "boto3 style" (list of dicts) format, rather than the normal simple dictionary.

We often just convert the resource objects the APIs return from CamelCase to snake_case, and add that as part of what the module returns.  When AWS suddenly adds support for Tags to a resource and changes what the API returns, we start returning the list-of-dict style tags, and need to go through a deprecation cycle before we can return the simple dict.

Since returning tags is new to this module, we can skip the "list-of-dict" and just return the "dict" format as "tags".

Reply:
(There are also some cases where reviews simply missed that the module originally returned the boto3 style tags)
- We're trying to tidy this up :)